src/device_info.sh

```diff
@@ -0,0 +1,173 @@
+# This file deals with functions related to hardware information
+
+include "$KW_LIB_DIR/remote.sh"
+include "$KW_LIB_DIR/kwlib.sh"
+
+declare -gA shared_data=(['ram']='' ['cpu_model']='' ['cpu_frequency']='' ['disks']='' ['os']='')
+
+# This function populates the ram element from the shared_data global variable
+# with the total RAM memory from the target machine in kB
+#
+# @target Target machine
+function get_ram()
+{
+  local target="$1"
+  local ram
+  local cmd
+
+  case "$target" in
+    1) # VM_TARGET
+      echo TODO
+      ;;
+    2) # LOCAL_TARGET
+      cmd="cat /proc/meminfo | grep 'MemTotal' | grep -o '[0-9]*'"
+      ram=$(cmd_manager "SILENT" "$cmd")
+      ;;
+    3) # REMOTE
+      echo TODO
+      ;;
+  esac
+  shared_data['ram']="$ram"
+}
+
+# This function provides the model and frequency of the CPU from a machine
+#
+# @target Target machine
+function get_cpu()
+{
+  local target="$1"
+  local cpu_model
+  local cpu_frequency
+  local cmd
+
+  case "$target" in
+    1) # VM_TARGET
+      echo TODO
+      ;;
+    2) # LOCAL_TARGET
+      cmd="lscpu | grep 'Model name:' | sed -r 's/Model name:\s+//g'"
+      cpu_model=$(cmd_manager 'SILENT' "$cmd")
+      cmd="lscpu | grep MHz | sed -r 's/(CPU.*)/\t\t\1/'"
```

Comment:
Hmmm... I'm not too fond of the idea of storing all of these CPU data in a single variable. It will be better if you have a key per option and only save the raw value.